### PR TITLE
fix: restructure determineRequestsReferrer to match better spec

### PR DESCRIPTION
**Pull request by Uzlopak**
While I investigated if there were significant changes in esbuild, I found the following change:

Screenshot from 2024-10-08 01-30-40

So the comments were added because they were of  `/** */`  notation and not of `//` notation. While I fixed, I saw that the implementation is not matching 1:1 the spec text, like the order of the cases. So I adapted it accordingly.

### CODE REVIEW

**Review by KhafraDev — APPROVED** [review]
Is it obvious that I didn't originally implement this? 😜

**Comment by Uzlopak** [comment]
Well atleast I get once i a while an approval for a fetch PR  😜

**Pull request merged**
